upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -618,24 +619,24 @@ func validateDNS(cluster *api.Cluster, cloud fi.Cloud) error {
 	zone := matches[0]
 	dnsName := strings.TrimSuffix(zone.Name(), ".")
 
-	glog.V(2).Infof("Doing DNS lookup to verify NS records for %q", dnsName)
-	ns, err := net.LookupNS(dnsName)
-	if err != nil {
-		return fmt.Errorf("error doing DNS lookup for NS records for %q: %v", dnsName, err)
-	}
+	if !fi.BoolValue(cluster.Spec.IgnoreNSCheck) {
```

Comment:
This is good, but it doesn't belong in the cluster spec.  Yes, this does mean you'll have to pass the flag every time.

A nicer option is probably explicit support for a private DNS HostedZone.  That didn't make sense before, but now we have topology=private, it does.